### PR TITLE
Sign In page

### PR DESCRIPTION
**Pull request by jdtzmn**
Screenshot

### With error:

Screenshot with error

### With patternfly error:

Screenshot with pattern fly error

### With patternfly layout

screen shot 2018-08-06 at 4 48 21 pm

### CODE REVIEW

File: src/components/SignIn/SignIn.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  }
+
+  hideError(e) {
+    if (!e.key || e.key === 'Enter') {
+      this.setState({ error: null });
+    }
+  }
+
+  render() {
+    if (this.state.toDashboard) {
```
Comment (jcaianirh) [inline src/components/SignIn/SignIn.js]: const {toDashboard, error, username, password, loading} = this.state;

File: src/components/SignIn/SignIn.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    this.handleUsername = this.handleUsername.bind(this);
+    this.handlePassword = this.handlePassword.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.showError = this.showError.bind(this);
+    this.hideError = this.hideError.bind(this);
+  }
+
+  componentWillMount() {
+    this.mounted = true;
```
Comment (jcaianirh) [inline src/components/SignIn/SignIn.js]: Whats going on here? Do we need to keep track of the mounted state?

File: src/components/SignIn/SignIn.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    if (this.state.toDashboard) {
+      return <Redirect to="/developers" />;
+    }
+
+    return (
+      <div className="signin">
+        <div className="signin-container">
+          {
+            this.state.error && (
+              <div className="signin-error-container">
```
Comment (jcaianirh) [inline src/components/SignIn/SignIn.js]: do you mind showing an image of the error case in the pr description please?

File: src/components/SignIn/SignIn.js (hunk 2 above)
Comment (jdtzmn) [reply src/components/SignIn/SignIn.js]: It has to do with memory leaks (very small ones of course) in React. The Promise will keep running (not easy to stop it), but there is no reason to re-render a nonexistent component when it redirects to the developers page. This was the simple fix I found.

File: src/components/SignIn/SignIn.js (hunk 3 above)
Comment (jdtzmn) [reply src/components/SignIn/SignIn.js]: Done!

File: src/components/SignIn/SignIn.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+  componentWillMount() {
+    this.mounted = true;
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  handleUsername(event) {
```
Comment (jcaianirh) [inline src/components/SignIn/SignIn.js]: onChange(event) {
  this.setState({ 
    [e.target.name]: e.target.value 
  });
//Add name to the form control 
}

File: src/components/SignIn/SignIn.js (hunk 3 above)
Comment (jcaianirh) [reply src/components/SignIn/SignIn.js]: Thanks @jdtzmn! Would you mind using the pfly react alert to handle these errors?

https://rawgit.com/patternfly/patternfly-react/gh-pages/index.html?selectedKind=patternfly-react%2FWidgets%2FAlert&selectedStory=Alert%20types&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs

Comment (jdtzmn) [reply src/components/SignIn/SignIn.js]: Done!

**Comment by jcaianirh** [comment]
I'd really like this to follow the patternfly pattern for Basic Login located here if possible: 

https://www.patternfly.org/pattern-library/application-framework/login-page/#code

We don't need to add the parts we don't plan on using (Language, Keep me logged in.., etc.).

**Comment by jdtzmn** [comment]
@jcaianirh Done!

**Pull request merged**